kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h

```diff
@@ -829,15 +827,15 @@ class ResidualBasedEliminationBuilderAndSolver
         if (A.size1() == 0 || BaseType::GetReshapeMatrixFlag() == true) //if the matrix is not initialized
         {
             A.resize(BaseType::mEquationSystemSize, BaseType::mEquationSystemSize, false);
-            ConstructMatrixStructure(pScheme, A, rElements, rConditions, CurrentProcessInfo);
+            ConstructMatrixStructure(pScheme, A, rModelPart);
         }
         else
         {
             if (A.size1() != BaseType::mEquationSystemSize || A.size2() != BaseType::mEquationSystemSize)
             {
                 KRATOS_WATCH("it should not come here!!!!!!!! ... this is SLOW");

```

Comment:
actually i think we should just throw an error here.

Reply:
OK 
I am adding the following error here 

KRATOS_ERROR <<"The equation system size has changed during the simulation. This is not permited."<<std::endl;